src/server/req_delete.c

```diff
@@ -362,230 +390,288 @@ req_deletejob(struct batch_request *preq)
+			} else if (subjobstate == JOB_STATE_LTR_EXPIRED) {
+				update_deletejob_stat(jid, preq, PBSE_NOHISTARRAYSUBJOB, &preply->brp_un.brp_delstat);
+				if (preq->rq_ind.rq_delete.tot_rpys == preq->rq_ind.rq_delete.tot_jobs) {
+					req_reject(PBSE_NOHISTARRAYSUBJOB, 0, preq);  
+					return;
+				} else
+					continue;
+			} else if ((pjob = parent->ji_ajtrk->tkm_tbl[offset].trk_psubjob)) {
+				/*
+				 * If the request is to also purge the history of the sub job then set ji_deletehistory to 1
+				 */
 				if (delhist)
 					pjob->ji_deletehistory = 1;
-				if (check_job_state(pjob, JOB_STATE_LTR_EXPIRED)) {
-					snprintf(log_buffer, sizeof(log_buffer),
-						msg_job_history_delete, preq->rq_user,
-						preq->rq_host);
-					log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_INFO,
-						pjob->ji_qs.ji_jobid,
-						log_buffer);
-					job_purge(pjob);
-				}else
-					dup_br_for_subjob(preq, pjob, req_deletejob2);
+				req_deletejob2(preq, pjob);
 			} else {
-				/* Queued, Waiting, Held, just set to expired */
-				if (sjst != JOB_STATE_LTR_EXPIRED) {
-					parent->ji_ajtrk->tkm_tbl[i].trk_substate = JOB_SUBSTATE_TERMINATED;
-					set_subjob_tblstate(parent, i, JOB_STATE_LTR_EXPIRED);
-					decr_single_subjob_usage(parent);
-				}
+				acct_del_write(jid, parent, preq, 0);
+				parent->ji_ajtrk->tkm_tbl[offset].trk_substate = JOB_SUBSTATE_TERMINATED;
+				set_subjob_tblstate(parent, offset, JOB_STATE_LTR_EXPIRED);
+				parent->ji_ajtrk->tkm_dsubjsct++;
+
+				decr_single_subjob_usage(parent);
+				preq->rq_ind.rq_delete.tot_rpys++;
+				if (preq->rq_ind.rq_delete.tot_rpys == preq->rq_ind.rq_delete.tot_jobs)
+					reply_ack(preq);
 			}
-		}
-		parent->ji_ajtrk->tkm_flags &= ~TKMFLG_NO_DELETE;
-
-
-		/* if deleting running subjobs, then just return;            */
-		/* parent will be deleted when last running subjob(s) ends   */
-		/* and reply will be sent to client when last delete is done */
-		/* If not deleteing running subjobs, delete2 to del parent   */
+			chk_array_doneness(parent);
+			continue;
 
-		if (--preq->rq_refct == 0) {
-			if ((parent = find_job(jid)) != NULL)
-				req_deletejob2(preq, parent);
-			else
-				reply_send(preq);
-		} else
-			acct_del_write(jid, parent, preq, 0);
-
-		return;
-	}
-	/* what's left to handle is a range of subjobs, foreach subjob 	*/
-	/* if running, do full delete, else just update state	        */
+		} else if (jt == IS_ARRAY_ArrayJob) {
+			/*
+			 * For array jobs the history is stored at the parent array level and also at the subjob level .
+			 * If the request is to delete the history of an array job then set  ji_deletehistory to 1 for
+			 * the parent array.The function chk_array_doneness() will take care of eventually
+			 *  purging the history .
+			 */
+			if (delhist)
+				parent->ji_deletehistory = 1;
+			/* The Array Job itself ... */
+			/* for each subjob that is running, delete it via req_deletejob2 */
+
+			++preq->rq_refct;
+			preq->rq_ind.rq_delete.tot_arr_jobs++;
```

Comment:
I don't see any usage of this variable other than this, can you please point me where its getting used?